File: 0921. Minimum Add to Make Parentheses Valid/main.js
/**
 * @param {string} s
 * @return {number}
 */
const minAddToMakeValid = function (s) {
  let stack = [];
  let res = 0;
  for (let c of s) {
    if (c === "(") {
      stack.push(c);
    } else {
      if (stack.length === 0) {
        res++;
      } else {
        stack.pop();
        // console.log(stack)
      }
    }
  }
  //   console.log(res);
  //   console.log(stack.length)
  return res + stack.length;
};

const s = "()";
console.log(minAddToMakeValid(s));

File: 0921. Minimum Add to Make Parentheses Valid/main.cpp
#include <algorithm>
#include <iostream>
#include <stack>
using namespace std;

/**
 * I: s = "())"
 * O: 1
 *
 * I: s = "((("
 * O: 3
 */

int minAddToMakeValid(string s);

int main(int argc, char const *argv[]) {
    string s = "()))((";
    cout << minAddToMakeValid(s) << endl;
    return 0;
}

int minAddToMakeValid(string s) {
    stack<char> st;
    int res = 0;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == '(') {
            st.push(s[i]);
        } else {
            if (st.empty()) {
                res++;
            } else {
                st.pop();
            }
        }
    }
    return res + int(st.size());
}
